src/Symfony/Component/Console/Application.php

```diff
@@ -466,7 +466,9 @@ public function get($name)
             $command = $this->commands[$name];
         } elseif ($this->commandLoader && $this->commandLoader->has($name)) {
             $command = $this->commandLoader->get($name);
-            $command->setName($name);
+            if (!$command->getName()) {
+                $command->setName($name);
+            }
             $this->add($command);
```

Comment:
Custom implementations of `add()` would be skipped for those and we would have to duplicate almost all of the `add()` logic, I think it's worth calling it